draft-ietf-tls-wkech.xml

```diff
@@ -397,8 +397,16 @@ Client <-->|                    | |                    |
     </t>
 
     <t>
-        [[TODO: What does the zone factory do if it encounters an unrecognized
-        field?]]
+        If the ZF does not recognise some of the JSON, e.g. when a new SvcParamKey is defined,
+        the ZF MUST NOT update the DNS. Such failures will not be directly visible to
```

Comment:
Strictly speaking, the JSON -> zone file conversion is mechanical even for unrecognized SvcParams.  The problem is in the zone file parser, not the zone factory.

```suggestion
        If the DNS server's zone file parser does not recognise some of the specified
        SvcParamKeys, it will not be able to load the updated zone file.  In this case,
        the ZF MUST NOT update the zone. Such failures will not be directly visible to
```

Reply:
Not sure of that one - that depends on how one implements the ZF (we don't know that presentation syntax will be used for example) and the problem with the JSON could be something else and not an unrecognised SvcParamKey so I'd say better to not make that change perhaps.

Reply:
OK, here's another version.

```suggestion
        If the ZF is unable to convert the JSON into a DNS zone (e.g., due to an
        unrecognized SvcParamKey), or if the resulting zone fails validation checks,
        the ZF MUST NOT update the DNS. Such failures will not be directly visible to
```

Reply:
Yep that one's good. (Bet a beer someone objects to "zone fails validation checks" later on:-)